src/event/mod.rs

```diff
@@ -640,4 +598,279 @@ mod test {
+                    Field(Quoted("bar baz".to_owned())),
+                    Field(Regular("a".to_owned())),
+                    Field(Regular("b".to_owned())),
+                ],
+                true.into(),
+                map![
+                    "foo":
+                        vec![
+                            Value::Null,
+                            Value::Null,
+                            map!["bar baz": map!["a": map!["b": true]],].into()
+                        ]
+                ],
+                Ok(()),
+            ),
+            (
+                map!["foo": vec![0, 1, 2]],
+                vec![Field(Regular("foo".to_owned())), Index(5)],
+                "baz".into(),
+                map![
+                    "foo":
+                        vec![
+                            0.into(),
+                            1.into(),
+                            2.into(),
+                            Value::Null,
+                            Value::Null,
+                            Value::from("baz"),
+                        ]
+                ],
+                Ok(()),
+            ),
+            (
+                map!["foo": "bar"],
+                vec![Field(Regular("foo".to_owned())), Index(0)],
+                "baz".into(),
+                map!["foo": vec!["baz"]],
+                Ok(()),
+            ),
+            (
+                map!["foo": Value::Array(vec![])],
+                vec![Field(Regular("foo".to_owned())), Index(0)],
+                "baz".into(),
+                map!["foo": vec!["baz"]],
+                Ok(()),
+            ),
+            (
+                map!["foo": Value::Array(vec![0.into()])],
+                vec![Field(Regular("foo".to_owned())), Index(0)],
+                "baz".into(),
+                map!["foo": vec!["baz"]],
+                Ok(()),
+            ),
+            (
+                map!["foo": Value::Array(vec![0.into(), 1.into()])],
+                vec![Field(Regular("foo".to_owned())), Index(0)],
+                "baz".into(),
+                map!["foo": Value::Array(vec!["baz".into(), 1.into()])],
+                Ok(()),
+            ),
+            (
+                map!["foo": Value::Array(vec![0.into(), 1.into()])],
+                vec![Field(Regular("foo".to_owned())), Index(1)],
+                "baz".into(),
+                map!["foo": Value::Array(vec![0.into(), "baz".into()])],
+                Ok(()),
+            ),
+        ];
+
+        for (object, segments, value, expect, result) in cases {
+            let object: BTreeMap<String, Value> = object;
+            let mut event = Event::from(object);
+            let expect = Event::from(expect);
+            let value: remap::Value = value;
+            let path = Path::new_unchecked(segments);
+
+            assert_eq!(event.insert(&path, value.clone()), result);
+            assert_eq!(event, expect);
+            assert_eq!(event.get(&path), Ok(Some(value)));
```

Comment:
I've noticed you tend to use this pattern often, during debugging I've noticed this can be hard to identify the case and intent of the test. Do you think we could help future authors by better describing the cornercases being tested when they are not obvious?